src/api/api.js

```diff
@@ -64,6 +66,22 @@ class Api {
 
     this.socket.onclose = () => {
       console.log("[socket.close]");
+      EventEmitter.emit("onDisconnect");
```

Comment:
this is fine

but what If a user does logout intentionally - will it run re-connect as well?